test/unit/prepare.test.ts

```diff
@@ -13,11 +13,20 @@ describe("readAndWriteNewCabal", () => {
     await writeFile(fakeCabalPath, cabalContent, "utf8");
   });
 
-  it("updates the version in the cabal file fixture", async () => {
+  it("updates the version in the cabal file fixture when version is semantic", async () => {
```

Comment:
The idea of BDD is to be able to group tests by scenarios. Adding a "when ..." in the `it(..)` function defeats that purpose. You can use `context(..)` to group scenarios instead:
```ts
context("when version is semantic", () => {
  it("updates the version in the cabal file fixture", () => {
    // ...rest of the test 
  });
});

context("when version is not semantic", () => {
  it("updates the version in the cabal file fixture with the given prefix", () => {
    // ...rest of the test
  });
});
```